### PR TITLE
4792 report versioning schema changes

### LINKED ISSUE #4792: Reports-versioning: Schema Changes
## Is your feature request related to a problem? Please describe 👀

To handle report versioning, we need to have some additional fields in the reports table in open-mSupply.

**Schema Changes**
is_custom boolean
version text field
code text field

## Describe the solution you'd like 🎁
New fields as above.

- [ ] Migration
- [ ] Repo layer
- [ ] tests pass in Postgres and Sqlite

Don't need to update graphql, service or sync yet...

### PR DESCRIPTION
**Pull request by fergie-nz**
Fixes #4792 

# 👩🏻‍💻 What does this PR do?

Adds schema changes for report versioning

## 💌 Any notes for the reviewer?

Has dummy values for report sync translation from OG - but we will want to remove this in future. 

# 🧪 Testing

No testing required - no user changes

# 📃 Documentation

[**Part of an epic**:](https://github.com/msupply-foundation/open-msupply/issues/4800) which will have documentation included later

### CODE REVIEW

File: server/service/src/sync/translations/report.rs
Comment (jmbrunskill) [inline server/service/src/sync/translations/report.rs:126]:
```suggestion
            is_custom: true,
            version: 1.0.to_string(),
            code: id,
```
I don't think it's a good idea to check in code to develop with todo!() like this as it breaks tests which could cause people to spend time trying to fix, and/or we might ignore for ever.

Not a major, `todo!` is much better than putting something wrong it at least!
But I think we know what to put in now, so lets do that now... IMHO

File: server/repository/src/db_diesel/report_row.rs
Comment (jmbrunskill) [inline server/repository/src/db_diesel/report_row.rs:51]:
```suggestion
      code -> Text,
```
Lets go with not null?

File: server/repository/src/db_diesel/report_row.rs
Comment (jmbrunskill) [inline server/repository/src/db_diesel/report_row.rs:75]:
```suggestion
    pub code: String,
```

File: server/repository/src/db_diesel/report_row.rs
Comment (jmbrunskill) [inline server/repository/src/db_diesel/report_row.rs:89]:
```suggestion
            is_custom: true,
```

File: server/service/src/report/report_service.rs
Comment (jmbrunskill) [inline server/service/src/report/report_service.rs:707]:
```suggestion
            is_custom: true,
            version: "1.0".to_string(),
            code: "report_1".to_string(),
```

File: server/service/src/report/report_service.rs
Comment (jmbrunskill) [inline server/service/src/report/report_service.rs:721]:
```suggestion
            is_custom: true,
            version: "1.0".to_string(),
            code: "report_base_1".to_string(),
```

File: server/service/src/sync/test/test_data/report.rs
Comment (jmbrunskill) [inline server/service/src/sync/test/test_data/report.rs:44]:
```suggestion
            code: REPORT_1.0.to_string(), // for now any thing sync'd from mSupply has the same id as code
```

File: server/cli/src/cli.rs
Comment (jmbrunskill) [inline server/cli/src/cli.rs:447]: I think we're probably ok to leave this as todo! as we're about to update the logic, but normally wouldn't want to have todo!() in main...

**Review by jmbrunskill — CHANGES_REQUESTED** [review]
Thanks Fergus. SOrry for the long wait on the review!

It's all looking good to me, I've had a think about defaults and I think we're best to put defaults in for all of the new fields.

Have made suggestions, can you apply them and make sure everything compiles and tests correctly before we merge?

File: server/repository/src/migrations/v2_03_00/add_report_versioning.rs
Comment (fergie-nz) [reply server/repository/src/migrations/v2_03_00/add_report_versioning.rs:16]: Nice! Thanks James, sounds good.

I'm wondering about the specific use case where we might want to override a standard report with a custom report - would having an old standard report listed as a custom report be annoying in this case?

**Comment by fergie-nz** [comment]
All tests passing now - though the intermittent failing of some tests in develop is increasingly making it awkward to check if tests are passing okay!

**Review by jmbrunskill — APPROVED** [review]
Looks good now, thanks Fergus!

**Pull request merged**
